addons/udes_stock_picking_batch/views/stock_picking_batch.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0"?>
+<odoo>
+    <data>
+
+        <!-- Stock picking batch form -->
+        <record id="stock_picking_batch_form" model="ir.ui.view">
+            <field name="name">udes_stock.stock.picking.batch.form</field>
+            <field name="model">stock.picking.batch</field>
+            <field name="inherit_id"
+                   ref="stock_picking_batch.stock_picking_batch_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='state']" position="replace">
+                    <field name="state" widget="statusbar" statusbar_visible="waiting,ready,in_progress,done"/>
+                </xpath>
+                <xpath expr="//button[@name='action_confirm']" position="after">
+                    <button name="mark_as_todo" string="Mark as Todo"
+                            confirm="Are you sure you want to mark this batch as todo?" type="object" states="draft"/>
+                </xpath>
+                <xpath expr="//button[@name='action_cancel']" position="replace">
+                    <button name="action_cancel" string="Cancel" type="object" states="ready,waiting,in_progress"/>
+                </xpath>
+            </field>
+        </record>
+
+        <!-- Stock picking batch filter -->
+        <record id="stock_picking_batch_filter" model="ir.ui.view">
+            <field name="name">udes_stock.stock.picking.batch.filter</field>
+            <field name="model">stock.picking.batch</field>
+            <field name="inherit_id"
+                   ref="stock_picking_batch.stock_picking_batch_filter"/>
+            <field name="arch" type="xml">
+                <xpath expr="//filter[@name='draft']" position="after">
+                    <filter name="ready" string="Ready"
+                            domain="[('state','=', 'ready')]"
+                            help="Batch Pickings Ready"/>
+                    <filter name="waiting" string="Waiting"
+                            domain="[('state','=', 'waiting')]"
+                            help="Batch Pickings Waiting"/>
+                </xpath>
+                <xpath expr="//filter[@name='in_progress']" position="replace">
+                    <filter name="in_progress" string="In Progress" domain="[('state', '=', 'in_progress')]"
+                            help="Batch Transfers not finished"/>
+                </xpath>
+                <xpath expr="//filter[@name='in_progress']" position="after">
+                    <filter name="cancel" string="Cancelled"
```

Comment:
Q: Does this make the filter show at the bottom of all the states?

Reply:
Could we put Cancelled underneath done?

It's not an expected state in the "happy path" flow if that makes ense. :)